llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -366,6 +369,18 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
   }
   case TargetOpcode::G_SELECT:
     return selectSelect(MI, MIB, MRI);
+  case TargetOpcode::G_FCMP: {
+    CmpInst::Predicate Pred =
+        static_cast<CmpInst::Predicate>(MI.getOperand(1).getPredicate());
+
+    if (!emitFPCompare(Pred, MI.getOperand(0).getReg(),
+                       MI.getOperand(2).getReg(), MI.getOperand(3).getReg(),
+                       MIB))
+      return false;
+
+    MI.eraseFromParent();
```

Comment:
Should we move the `eraseFromParent` into `emitFPCompare` like the other select functions do?